### ISSUE TITLE
WiFi problems after installing gsconnect

### ISSUE THREAD

**Issue by sschuhmann** [andyholmes/gnome-shell-extension-gsconnect#530]
After installing gsconnect my wifi adapter always gets renamed after connecting to a network. 

Distro: Antergos

**Comment by andyholmes** [comment]
Thanks for reporting. Can you please provide:

* GNOME Shell version
* GSConnect version
* Explanation or screenshot of what a "renamed adapter" is

**Comment by sschuhmann** [comment]
Gnome Shell Version: *3.32*
GSConnect Version: *21-2* ()

The network interface gets a new name assigned after connecting to a wifi network. which leads to a disconnection. E.g. wlp01 gets a new name assigned --> wlp01-01

**Comment by andyholmes** [comment]
As far as I know there is no way for GSConnect to rename a system device in that way, and it certainly doesn't do so on purpose.

Can you please generate a log as described in [Generate Support Log](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Help#generate-support-log)?

**Comment by nicologreggio** [comment]
Hi, I'm posting here because my problem seems to be related. I'm experiencing wifi connection problems since I installed GSConnect. They start when i try to do certain operation, like mounting the phone file systems, or when configuring something in GSConnect settings. What happens is that my computer lose connection to the wifi, the icon on the gnome top bar disappears for a few seconds, if I open the upper right menu I can see that it has switched to airplane mode, then after a few seconds airplane mode disappears, the wifi icon appears again, this time with a question mark (?), then it disappears again and after a few seconds it gets normal and I'm connected to network again. It's pretty annoying because it freezes the entire shell, and prevents me from doing other things, and I couldn't figure out what may be causing the problem.
The wifi disconnecting also happens when I'm not doing anything with GSConnect, and I'm forced to disabled it, that is the only way I can keep a stable connection. I had to disable the extension while i was writing this, because eveything had become unresponsive
I'm linking the paste to the log generated by "Generate Support Log" and attaching a picture of the system logs where it shows repetitive and consequent problems with the wifi
 [Generated Support Log related to wifi problems](https://pastebin.com/4LG5LAdZ)

![LOG](https://user-images.githubusercontent.com/25263073/57968501-b2d0ca00-796b-11e9-8eca-be44065a5a00.png)

**Comment by andyholmes** [comment]
Hi, it's hard to tell what's happening with GSConnect since there's no mention of it doing anything in your logs. There have been report of KDE Connect broadcasts causing problems with some WiFi repeaters; are you using on of those?

**Comment by nicologreggio** [comment]
Sorry, I didn't get what you meant, what do you mean by "KDE Connect broadcasts"?

**Comment by andyholmes** [comment]
KDE Connect clients identify themselves by broadcasting to the network over port 1716. There have been some reports of these [interfering with some equipment](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Help#interference-with-other-equipment).

**Comment by nicologreggio** [comment]
I read all the issues and the things related to what you linked. I'm not experiencing problem with other devices on the network (which is just a simple home setup with 2 wifi router wired together to cover distance), only on the pc where GSConnect is installed. And I remember it used to work a few months ago, then I stopped using it for a little because I changed various distros, now I reinstalled it and it has started causing problems.

**Comment by andyholmes** [comment]
Unfortunately there's no messages in your log related to GSConnect. Since GSConnect doesn't do anything but open TCP streams and send UDP broadcasts, I just don't know what could be causing this.

Your log says:

    May 18 12:28:02 kernel: iwlwifi 0000:03:00.0: Microcode SW error detected.  Restarting 0x82000000.
    -- snip --
    May 18 12:28:02 kernel: ieee80211 phy0: Hardware restart was requested

This seems to indicate there is some bug in your driver, possibly triggered by GSConnect, but as I said GSConnect does nothing special. It's only opens TCP connections and sends UDP broadcasts. Unless there is something special about your network like a proxy, VPN or something else I'm not sure where to go from here.

**Comment by ferdnyc** [comment]
@nicologreggio The log you posted shows that your wireless driver (iwlmvm) is crashing in the wireless device firmware, specifically the `TDLS_CONFIG_CMD`:

```
May 18 12:28:02 kernel: iwlwifi 0000:03:00.0: FW error in SYNC CMD TDLS_CONFIG_CMD
```

[TDLS](https://en.wikipedia.org/wiki/TDLS), AKA 802.11z, is a protocol for directly connecting two devices on the same wireless network for the purposes of sending data between them. 

It's _possible_ GSConnect is triggering this crash only because it causes your Ubuntu system and your Android device to communicate directly with each other (when they otherwise would not), and therefore they attempt to establish a TDLS link. But if so, that's merely a side effect of GSConnect talking to the Android device; as Andy said, GSConnect doesn't do anything special in that regard.

Problems caused by TDLS are [apparently not unheard of](https://bugzilla.kernel.org/show_bug.cgi?id=199971). One suggestion might be to update your system to a more recent linux kernel (5.0.0 is pretty outdated already), and possibly update the `linux-firmware` and/or `nic-firmware` packages that supply the wireless firmware as well. Any of those might hopefully contain fixes for this bug. If that doesn't work... I'm not sure whether there's a way to disable TDLS, I'm not immediately finding one.

**Comment by nicologreggio** [comment]
@ferdnyc thank you for the explanation, I wasn't aware of this at all. I will try what you suggested.

**Comment by ferdnyc** [comment]
Yeah, honestly I can fake it like I know what I'm talking about, but this is the first I've heard of it either. My home network is still 802.11n. :lying_face:  But that's also how I can say with pretty strong confidence that nothing in the GSConnect code is in any way directly related to TDLS. As with the user in that kernel bug report I linked to, the problem appears to merely be a side-effect of communication between two devices on the same network, both of which support TDLS (in theory) and therefore _attempt_ to use it, with less-than-desirable results.

This [regulatory declaration for your laptop model](https://dlcdnets.asus.com/pub/ASUS/nb/X580VD/CE_Doc_X580V_EN.pdf) — visible in your log dump — indicates that the wireless adapter is probably an Intel 8265.NGW device, which would use one of the `iwlwifi-8265-<something>.ucode` firmware files from your system. (Exactly what the value of `<something>` would be depends on the exact hardware revision.) 

Let's see, on your Antergos distro — which [just announced its termination yesterday, ouch!](https://antergos.com/blog/antergos-linux-project-ends/) — on your _Arch_ distro, then, that'd be in the `/usr/lib/firmware/` directory, as installed by the [`linux-firmware` package](https://www.archlinux.org/packages/core/any/linux-firmware/), which is updated in AUR pretty regularly so it should have all of the latest updates. The 8265 WiFi firmware was last updated in December of last year, since then it's only the Bluetooth code that's been touched. So, building from the AUR package should get you everything that the master kernel tree has available.

In the end, that other bug report for that other user was solved by the kernel developers simply having them install what they call the ["backport driver"](https://wireless.wiki.kernel.org/en/users/drivers/iwlwifi/core_release#how_to_install_the_driver), which is a newer version of the iwlwifi kernel driver direct from Intel, containing fixes not (yet) present in their distro's kernels. But if you're using an Arch-based distro, hopefully the latest and greatest driver versions should be available as a `linux` package update, hopefully just a simple `sudo {yay,aurman}` command away. (Or replace with your package manager of choice...)

If _**not**_, then that backport driver page can hopefully offer some resources.
